### ISSUE TITLE
Immutable Builder from GsonAdapter

### ISSUE THREAD

**Issue by pushbit** [immutables/immutables#615]
I'm not sure about the interface for this, but it would be nice if we could somehow get the Immutable Builder from a GsonAdapter instead of only the built Immutable.  Sometimes I have Immutables that contain the JSON fields plus additional fields which I would like to fill in after the JSON data is populated.

**Comment by elucash** [comment]
Can you make those fields nullable or optional and then set them using `with***` methods or `Builder.from(instance).***.build()` afterwards?

**Comment by pushbit** [comment]
Yes, those are possible workarounds, though ideally I would like to be able to keep logically non-null attributes @NonNull and get the original builder to avoid the build-create-build-create thrash.

**Comment by elucash** [comment]
While it might be a good idea to generate adapters for builders, it seems too much of an effort, not to say that huge one but requires some extensive copy/paste/edit/refactor cycle. Right now it cannot be on the top of our priorities, but we're open for PR.
